File: frontend/src/constants/produtos.js
import React from "react";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export const TIPOS_PRODUTO = {
  "cartao-de-credito": "cartao-de-credito",
  seguro: "seguro",
  consorcio: "consorcio",
};

export const PRODUTOS = {
  "cartao-de-credito": {
    nome: "Cartão de Crédito",
    icone: <FontAwesomeIcon icon="credit-card" className="product-icon" />,
    id: "cartao-de-credito",
    ordem: 1,
    tipo_produto: TIPOS_PRODUTO["cartao-de-credito"],
  },
  "seguro-vida": {
    nome: "Seguro de Vida Individual",
    icone: <FontAwesomeIcon icon="user" className="product-icon" />,
    id: "seguro-vida",
    ordem: 2,
    tipo_produto: TIPOS_PRODUTO.seguro,
  },
  "seguro-residencial": {
    nome: "Seguro Residencial",
    icone: <FontAwesomeIcon icon="house-user" className="product-icon" />,
    id: "seguro-residencial",
    ordem: 3,
    tipo_produto: TIPOS_PRODUTO.seguro,
  },
  "seguro-automotivo": {
    nome: "Seguro Automotivo",
    icone: <FontAwesomeIcon icon="car" className="product-icon" />,
    id: "seguro-automotivo",
    ordem: 4,
    tipo_produto: TIPOS_PRODUTO.seguro,
  },
  consorcio: {
    nome: "Consórcio",
    icone: (
      <>
        <span className="fa-layers fa-fw">
          {/**
           * Transformações base aplicadas:
           * car: shrink-5 down-5 left-5
           * slash: shrink-5
           * home: shrink-5 up-5 right-7
           *
           * Depois foi aplicada + left-5 para centralizar o conjunto
           */}
          <FontAwesomeIcon
            icon="car"
            transform="shrink-5 down-5 left-10"
            className="product-icon"
          />
          <FontAwesomeIcon
            icon="slash"
            className="product-icon"
            transform="shrink-5 left-5"
          />
          <FontAwesomeIcon
            icon="home"
            transform="shrink-5 up-5 right-2"
            className="product-icon"
          />
        </span>
      </>
    ),
    id: "consorcio",
    ordem: 5,
    tipo_produto: TIPOS_PRODUTO.consorcio,
  },
};

File: backend/backoffice/models/produtos.py
import enum

import dataclasses


class TIPO_PRODUTO(str, enum.Enum):
    """
    Enum que controla o tipo do produto.

    Usado para pegar o conjunto de transições de status correspondente
    """

    CARTAO_DE_CREDITO = "cartao-de-credito"
    SEGURO = "seguro"
    CONSORCIO = "consorcio"


@dataclasses.dataclass(frozen=True)
class Produto:
    tipo_produto: TIPO_PRODUTO
    nome: str


PRODUTOS = {
    "cartao-de-credito": Produto(
        nome="Cartão de Crédito", tipo_produto=TIPO_PRODUTO.CARTAO_DE_CREDITO
    ),
    "seguro-vida": Produto(nome="Seguro de Vida", tipo_produto=TIPO_PRODUTO.SEGURO),
    "seguro-residencial": Produto(
        nome="Seguro Residencial", tipo_produto=TIPO_PRODUTO.SEGURO
    ),
    "seguro-automotivo": Produto(
        nome="Seguro Automotivo", tipo_produto=TIPO_PRODUTO.SEGURO
    ),
    "consorcio": Produto(nome="Consórcio", tipo_produto=TIPO_PRODUTO.CONSORCIO),
}
